pyonfx/timestamps.py

```diff
@@ -0,0 +1,555 @@
+                    "If you specify a value for ``fpms``, you cannot specify a value for ``last_frame_time``"
+                )
+
+            self.fpms = fpms
+            self.timestamps = [0]
+            self.last_frame_time = Fraction(0)
+
+    @property
+    def rounding_method(self) -> RoundingMethod:
+        return self.__rounding_method
+
+    @rounding_method.setter
+    def rounding_method(self, value: Any):
+        raise AttributeError("You cannot change the value of rounding_method")
+
+    @classmethod
+    def from_fps(
+        cls: Timestamps,
+        fps: Union[int, float, Fraction, Decimal],
+        rounding_method: Optional[RoundingMethod] = RoundingMethod.ROUND,
+    ) -> Timestamps:
+        """Create timestamps based on the `fps` provided.
+
+        Parameters:
+            fps (positive int, int | float | Fraction | Decimal): Frames per second.
+            rounding_method (RoundingMethod, optional): A rounding method. See the comment in Timestamps description about floor vs round.
+
+        Returns:
+            A Timestamps instance.
+        """
+        if not 0 < fps <= 1000:
+            raise ValueError(
+                "Parameter ``fps`` must be between 0 and 1000 (0 not included)."
+            )
+
+        fpms = Fraction(fps) / Fraction(1000)
+
+        timestamps = cls(
+            rounding_method=rounding_method,
+            fpms=fpms,
+        )
+        return timestamps
+
+    @classmethod
+    def from_timestamps_file(
+        cls: Timestamps,
+        path_to_timestamps_file_or_content: Union[str, os.PathLike[str]],
+        normalize: Optional[bool] = True,
+        rounding_method: Optional[RoundingMethod] = RoundingMethod.ROUND,
+    ) -> Timestamps:
+        """Create timestamps based on a [timestamps file](https://mkvtoolnix.download/doc/mkvmerge.html#mkvmerge.external_timestamp_files).
+
+        To extract the timestamps file, you have 2 options:
+            - Open the video with Aegisub. "Video" --> "Save Timecodes File";
+            - Using [gMKVExtractGUI](https://sourceforge.net/projects/gmkvextractgui/) (warning: it will produce one timestamp too many at the end of the file, and you will need to manually remove it).
+
+        Parameters:
+            path_to_timestamps_file_or_content (str | os.PathLike[str]):
+                Path for the timestamps file (either relative to your .py file or absolute).
+                Or, it can directly be a string of the timestamps file content.
+            normalize (bool, optional): If True, it will shift the timestamps to make them start from 0. If false, the option does nothing.
+            rounding_method (RoundingMethod, optional): A rounding method. See the comment in Timestamps description about floor vs round.
+
+        Returns:
+            A Timestamps instance.
+        """
+
+        if os.path.isfile(path_to_timestamps_file_or_content):
+            with open(path_to_timestamps_file_or_content, "r") as f:
```

Comment:
We may need to specify the encoding to "utf-8" because timestamps can contains comments with non-ascii character.

[mkvtoolnix](https://gitlab.com/mbunkus/mkvtoolnix/-/blob/c6911337e4019702eff27758f89ee47a88b8f269/src/common/mm_io.cpp#L51-71) doesn't seems to care about the encoding. It will just convert the bytes to char, so equivalent to ascii.

But if we set the encoding to "ascii", it will just raise an exception when the file contains a non-ascii char, so it would be safer to use utf-8

What do you think about it?